### PR TITLE
Async broadcaster publisher

### PR DESCRIPTION
**Pull request by GaryWilber**
Make the `IPublisher` `emit` method async. 

This will allow changing the emit call to wait for the redis publish to complete. `ioredis` returns promises for all it's redis command methods. Waiting for the publish command to finish before processing the next batch should be more efficient because it will allow for batching more messages into each individual command

### CODE REVIEW

**Comment by hedasilv** [comment]
How would this affect the system if we use `node-redis` instead of `ioredis`?

**Comment by GaryWilber** [comment]
> How would this affect the system if we use `node-redis` instead of `ioredis`?

It doesn't really do anything in that case. The emit promise will immediately resolve.

File: server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts [hunk 1]
```diff
@@ -113,23 +113,32 @@ export class BroadcasterLambda implements IPartitionLambda {
...
                 }
-            });
+            }
+
+            try {
+                await Promise.all(promises);
+            } catch (ex) {
+                this.context.error(ex, { restart: true });
+                return;
+            }
```
Comment (tanviraumi) [inline server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts:139]: Should we refactor it this way?

if (this.publisher.emit) {
  for (const [topic, batch] of this.current) {
    // Do all promise stuff
  }
} else {
    for (const [topic, batch] of this.current) {
      // do regular stuff
  }
}

Comment (GaryWilber) [reply server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts:139]: Yea that sounds good

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**